tmt/steps/provision/bootc.py

```diff
@@ -252,6 +253,8 @@ def _build_derived_image(self, base_image: str) -> str:
         """
         Build a "derived" container image from the base image with tmt dependencies added
         """
+        if self.is_dry_run:
+            return ''
```

Comment:
I dare to say that `return ''` is not the right move: if it makes no sense to "build derived image", as suggested by the empty string signalling "nope", then whoever called this method should not have done so. Returning an empty string makes it look like the method run, but the output is dummy and harder for linters to distinguish from a regular outcome.